asyncpg/protocol/record/recordobj.c

```diff
@@ -77,20 +159,26 @@ ApgRecord_New(PyTypeObject *type, PyObject *desc, Py_ssize_t size)
     }
 
     Py_INCREF(desc);
-    o->desc = (ApgRecordDescObject*)desc;
+    o->desc = (ApgRecordDescObject *)desc;
     o->self_hash = -1;
     if (need_gc_track) {
         PyObject_GC_Track(o);
     }
-    return (PyObject *) o;
+    return (PyObject *)o;
 }
 
-
 static void
-record_dealloc(ApgRecordObject *o)
+record_dealloc(PyObject *self)
 {
+    ApgRecordObject *o = (ApgRecordObject *)self;
     Py_ssize_t i;
     Py_ssize_t len = Py_SIZE(o);
+    record_module_state *state;
+
+    state = find_module_state_by_def(Py_TYPE(o));
+    if (state == NULL) {
+        return;
+    }
```

Comment:
Should this be an `assert(state);`?
Could this ever happen for non-bug reasons?

Reply:
(If so, let's document them here)